src/Module/Post/Edit.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Post;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Content\Feature;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Model\Contact;
+use Friendica\Model\Post;
+use Friendica\Model\User;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Network\HTTPException;
+use Friendica\Util\Crypto;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Controller to edit a post
+ */
+class Edit extends BaseModule
+{
+	/** @var IHandleUserSessions */
+	protected $session;
+	/** @var SystemMessages */
+	protected $sysMessages;
+	/** @var App\Page */
+	protected $page;
+	/** @var App\Mode */
+	protected $mode;
+	/** @var App */
+	protected $app;
+	/** @var bool */
+	protected $isModal = false;
+
+	public function __construct(L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, IHandleUserSessions $session, SystemMessages $sysMessages, App\Page $page, App\Mode $mode, App $app, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->session     = $session;
+		$this->sysMessages = $sysMessages;
+		$this->page        = $page;
+		$this->mode        = $mode;
+		$this->app         = $app;
+	}
+
+
+	protected function content(array $request = []): string
+	{
+		$this->isModal = $request['mode'] ?? '' === 'none';
+
+		if (!$this->session->getLocalUserId()) {
+			$this->errorExit($this->t('Permission denied.'), HTTPException\UnauthorizedException::class);
+		}
+
+		$postId = $this->parameters['post_id'];
```

Comment:
Shouldn't we always guard against missing parameters?
```suggestion
		$postId = $this->parameters['post_id'] ?? 0;
```